### PR TITLE
ARUHA-141 Support definition of retention time

### PR DESCRIPTION
**Pull request by adyach**

### CODE REVIEW

File: src/main/java/de/zalando/aruha/nakadi/domain/EventTypeOptions.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package de.zalando.aruha.nakadi.domain;
+
+import javax.annotation.Nullable;
+
+public class EventTypeOptions {
+
+    @Nullable
+    private Long retentionTime;
```
Comment (v1ctor) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventTypeOptions.java]: I think we are almost adopted usage of Optional

File: src/main/java/de/zalando/aruha/nakadi/domain/EventType.java [hunk 2]
```diff
@@ -49,6 +51,10 @@
     @Nullable
     private EventTypeStatistics defaultStatistic;
 
+    @Valid
+    @Nullable
+    private EventTypeOptions options;
```
Comment (v1ctor) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventType.java]: What is options? Do we have to move another fields there?

Comment (adyach) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventType.java]: https://techjira.zalando.net/browse/ARUHA-141

Comment (v1ctor) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventType.java]: Ok. But isn't an answer :) I don't understand what should belong to options and what shouldn't.

Comment (adyach) [inline src/main/java/de/zalando/aruha/nakadi/domain/EventType.java]: I agree that is not the clearest name for such field. I think it is more about internal Kafka adjustments and other properties which are related to internals of Nakadi.

**Comment by v1ctor** [comment]
👍

File: api/nakadi-event-bus-api.yaml [hunk 3]
```diff
@@ -1084,6 +1090,16 @@ definitions:
       - read_parallelism
       - write_parallelism
 
+  EventTypeOptions:
+    type: object
+    description: |
+      Additional parameters for tuning internal behavior of Nakadi.
+    properties:
+      retention_time:
+        type: long
```
Comment (rcillo) [inline api/nakadi-event-bus-api.yaml]: Are you sure it's possible to use a type `long`? I thought that we should use `type: integer` and `format: int64` https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -24,8 +24,11 @@ spring:
     validationQuery: SELECT 1
 nakadi:
   topic:
+    min:
+      retentionMs: 86400000 # 1 days
     max:
       partitionNum: 8
+      retentionMs: 172800000 # 2 days
```
Comment (rcillo) [inline src/main/resources/application.yml]: I thought we would be using a larger `max` value. Teams are complaining that if they have a problem during the weekend, when they are back to work on monday events are already gone. We should have at least 3 days, so they have a change to fix it on Monday.

**Pull request merged**